site/src/pages/UsersPage/UsersPageView.tsx

```diff
@@ -60,54 +54,38 @@ export const UsersPageView: FC<React.PropsWithChildren<UsersPageViewProps>> = ({
   isNonInitialPage,
   actorID,
   authMethods,
-  count,
-  limit,
-  onPageChange,
-  page,
   groupsByUserId,
+  usersQuery,
 }) => {
   return (
     <>
       <UsersFilter {...filterProps} />
 
-      <TableToolbar>
-        <PaginationStatus
-          isLoading={Boolean(isLoading)}
-          showing={users?.length ?? 0}
-          total={count ?? 0}
-          label="users"
-        />
-      </TableToolbar>
-
-      <UsersTable
-        users={users}
-        roles={roles}
-        groupsByUserId={groupsByUserId}
-        onSuspendUser={onSuspendUser}
-        onDeleteUser={onDeleteUser}
-        onListWorkspaces={onListWorkspaces}
-        onViewActivity={onViewActivity}
-        onActivateUser={onActivateUser}
-        onResetUserPassword={onResetUserPassword}
-        onUpdateUserRoles={onUpdateUserRoles}
-        isUpdatingUserRoles={isUpdatingUserRoles}
-        canEditUsers={canEditUsers}
-        oidcRoleSyncEnabled={oidcRoleSyncEnabled}
-        canViewActivity={canViewActivity}
-        isLoading={isLoading}
-        isNonInitialPage={isNonInitialPage}
-        actorID={actorID}
-        authMethods={authMethods}
-      />
-
-      {count !== undefined && (
-        <PaginationWidgetBase
-          totalRecords={count}
-          pageSize={limit}
-          onPageChange={onPageChange}
-          currentPage={page}
+      <PaginationContainer
+        paginationResult={usersQuery}
+        paginationUnitLabel="users"
```

Comment:
Nit: Since the component is already a Pagination thing I don't think we should prefix the props with `pagination` because the component name makes it clear enough IMO.

Reply:
Yeah, I can agree. I can rename it